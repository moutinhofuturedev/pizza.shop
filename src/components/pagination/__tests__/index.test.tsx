import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'

import { Pagination } from '../index'

const onPageChangeCallback = vi.fn()

describe('<Pagination />', () => {
  beforeEach(() => {
    onPageChangeCallback.mockClear()
  })

  it('should display the right amount of pages and results', () => {
    const { container } = render(
      <Pagination
        totalCount={200}
        perPage={10}
        pageIndex={0}
        onPageChange={onPageChangeCallback}
      />
    )

    const totalResults = screen.getByText('Total de 200 item(s)')
    const pageCount = screen.getByText('Página 1 de 20')

    expect(pageCount).toBeInTheDocument()
    expect(totalResults).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })

  it('should be able to navigate to the next page', async () => {
    const wrapper = render(
      <Pagination
        pageIndex={0}
        totalCount={200}
        perPage={10}
        onPageChange={onPageChangeCallback}
      />
    )

    const nextPageButton = wrapper.getByRole('button', {
      name: 'Próxima página',
    })

    await userEvent.click(nextPageButton)

    expect(onPageChangeCallback).toHaveBeenCalledWith(1)
  })

  it('should be able to navigate to the previous page', async () => {
    const wrapper = render(
      <Pagination
        pageIndex={1}
        totalCount={200}
        perPage={10}
        onPageChange={onPageChangeCallback}
      />
    )

    const previousPageButton = wrapper.getByRole('button', {
      name: 'Página anterior',
    })

    await userEvent.click(previousPageButton)

    expect(onPageChangeCallback).toHaveBeenCalledWith(0)
  })

  it('should be able to navigate to the first page', async () => {
    const wrapper = render(
      <Pagination
        pageIndex={1}
        totalCount={200}
        perPage={10}
        onPageChange={onPageChangeCallback}
      />
    )

    const firstPageButton = wrapper.getByRole('button', {
      name: 'Primeira página',
    })

    await userEvent.click(firstPageButton)

    expect(onPageChangeCallback).toHaveBeenCalledWith(0)
  })

  it('should be able to navigate to the last page', async () => {
    const wrapper = render(
      <Pagination
        pageIndex={1}
        totalCount={200}
        perPage={10}
        onPageChange={onPageChangeCallback}
      />
    )

    const lastPageButton = wrapper.getByRole('button', {
      name: 'Última página',
    })

    await userEvent.click(lastPageButton)

    expect(onPageChangeCallback).toHaveBeenCalledWith(19)
  })
})
