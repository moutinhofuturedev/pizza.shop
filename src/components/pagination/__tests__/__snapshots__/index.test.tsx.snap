// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Pagination /> > should display the right amount of pages and results 1`] = `
<div>
  <div
    class="flex items-center justify-between"
  >
    <span
      class="text-sm text-muted-foreground"
    >
      Total de 
      200
       item(s)
    </span>
    <div
      class="flex items-center gap-6 lg:gap-8"
    >
      <div
        class="text-sm font-medium"
      >
        Página 
        1
         de 
        20
      </div>
      <div
        class="flex items-center gap-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
          disabled=""
        >
          <svg
            class="lucide lucide-chevrons-left h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m11 17-5-5 5-5"
            />
            <path
              d="m18 17-5-5 5-5"
            />
          </svg>
          <span
            class="sr-only"
          >
            Primeira página
          </span>
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
          disabled=""
        >
          <svg
            class="lucide lucide-chevron-left h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m15 18-6-6 6-6"
            />
          </svg>
          <span
            class="sr-only"
          >
            Página anterior
          </span>
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
        >
          <svg
            class="lucide lucide-chevron-right h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m9 18 6-6-6-6"
            />
          </svg>
          <span
            class="sr-only"
          >
            Próxima página
          </span>
        </button>
        <button
          class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-8 w-8 p-0"
        >
          <svg
            class="lucide lucide-chevrons-right h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 17 5-5-5-5"
            />
            <path
              d="m13 17 5-5-5-5"
            />
          </svg>
          <span
            class="sr-only"
          >
            Última página
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
